src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  Directive,
+  Input,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+
+import { AuthStorageService } from '../services/auth-storage.service';
+
+@Directive({
+  selector: '[cdAuthStorage]'
+})
+export class AuthStorageDirective {
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef
+  ) { }
+
+  @Input('cdAuthStorage') set cdAuthStorage(condition: AuthStorageService) {
+    if (condition.isLoggedIn()) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    } else {
+      this.viewContainer.clear();
+    }
```

Comment:
@epuertat thanks for your suggestions. This got real interesting fast ...

I ran into several issues. 

*cdPermissions='configOpt' (single permission, equivalent to check this.authStorageService.getPermissions().configOpt.read)

For this case, it seems angular expects variables as parameters by default. To avoid doing further typecasting I found it best to pass the permission value as a string 

```javascript
   *cdAuthStorage="'osd'"
 ```

*cdPermissions='[configOpt, osd, ...]' (list of permissions, all of them by default read permissions.

For this case, It wasn't possible to evaluate the type of condition passed to *cdAuthStorage so I used lodash to do the evaluations instead. Luckily it was already a dashboard dependency. Also, the array has to  contain strings too 

```javascript
  *cdAuthStorage="['osd', 'rgw', 'pools']"
```

*cdPermissions={configOpt: [read, update], osd: [read]} (object containing a more detailed permission policy).

For this case, the permissions array has to be strings too 

```javascript
*cdAuthStorage="{rgw: ['read', 'update'], osd: ['read']}
```

I used the cd-health component for testing. Just added this test HTML 

```html
<cd-info-card cardTitle="Display if string" class="cd-status-card" 
    *cdAuthStorage="'osd'">
      This displays because you passed in a string
    </cd-info-card>

    <cd-info-card cardTitle="Display if Array" class="cd-status-card"
      *cdAuthStorage="['configOpt', 'osd', 'rgw']">
      This displays because you passed in an Array
    </cd-info-card>

    <cd-info-card cardTitle="Display if Object" class="cd-status-card"
      *cdAuthStorage="{rgw: ['read', 'update'], osd: ['read']}">
      This displays because you passed in an object
    </cd-info-card>
  ```
  
 Make check failed initially due to a wrongly written test for the directive. Made changes to it but still in progress.

Reply:
That's great! I see the tests are passing now.